pkg/operator/ceph/csi/cluster_config.go

```diff
@@ -112,6 +113,16 @@ func updateCsiClusterConfig(curr, clusterKey string, newCsiClusterConfigEntry *C
 	// a lock is acquired for the update operation. So concurrent updates (rare event) will block and
 	// wait for the other update to complete. Monitors and Subvolumegroup will be updated
 	// independently and won't collide.
+	for i, centry := range cc {
```

Comment:
I think we can avoid requiring the clusterInfo param if we loop in the following way:
```go
if newCsiClusterConfigEntry != nil {
	for i, centry := range cc {
		if newCsiClusterConfigEntry.ClusterID == centry.Namespace {
			// update the mons
		}
	}
}
```
If newCsiClusterConfigEntry is an entry for a subvolume group or rados namespace, the ClusterID just won't match any namespaces. If it is for a cluster, it will match all the namespaces.

Reply:
`newCsiClusterConfigEntry` will not be sent in the with clusterID, `newCsiClusterConfigEntry.ClusterID` will remain empty in all the places. not sure about the check will work. If we need to make it work I need to send clusterID in other places where we are calling `SaveClusterConfig`. let me know if its fine i will make changes, as i need to test it manually it might take time

Reply:
Oh right, part of my PR had been to fix getting ClusterID always set. :) The way it is we can actually compare instead with clusterKey, right?
```go
if clusterKey == centry.Namespace {
```

Reply:
This should work with a couple of other changes. PTAL